doc/services/pm/device.rst

```diff
@@ -20,90 +27,153 @@ in power saving mode. This method allows saving power even when the CPU is
+        TURN_OFF [label=PM_DEVICE_ACTION_TURN_OFF];
+        TURN_ON [label=PM_DEVICE_ACTION_TURN_ON];
+        FORCE_SUSPEND [label=PM_DEVICE_ACTION_FORCE_SUSPEND];
 
-   When using :ref:`pm-system`, device transitions can be run from the idle thread.
-   As functions in this context cannot block, transitions that intend to use blocking
-   API's **must** check whether they can do so with :c:func:`k_can_yield`.
+	SUSPEND -> SUSPENDING -> SUSPENDED;
+	FORCE_SUSPEND -> SUSPENDING;
+	RESUME -> ACTIVE;
+	TURN_OFF -> OFF;
+	TURN_ON -> ACTIVE;
+    }
+
+Devices do not directly change states, this is entirely done by the power management
+subsystem.
 
 Device Model with Power Management Support
 ******************************************
 
-Drivers initialize the devices using macros. See :ref:`device_model_api` for
-details on how these macros are used. Use the DEVICE_DEFINE macro to initialize
-drivers providing power management support via the PM control function.
-One of the macro parameters is the pointer to the PM action callback.
-If the driver doesn't implement any power control operations, it can initialize
-the corresponding pointer with ``NULL``.
+Drivers initialize devices using macros. See :ref:`device_model_api` for
+details on how these macros are used. When a driver supports power management
+there are some additional steps that a driver must do. The very first step is to
+define the power management resources for the particular driver, this is done
+using :c:macro:`PM_DEVICE_DEFINE` or
+:c:macro:`PM_DEVICE_DT_DEFINE`. These macros create a resource with
+all information necessary by the power management subsystem.
+
+After the device power management resource be created, it is still necessary to
+bind it to the device driver. This last step is done using :c:macro:`PM_DEVICE_GET`
+or :c:macro:`PM_DEVICE_DT_GET` for field `pm_device` in macros used to define / initialize a device.
+
+The example below shows, including an action callback implementation, how to
+support power management on a device.
+
+.. code-block:: c
+
+    static int dummy_driver_pm_action(const struct device *dev,
+                               enum pm_device_action *action)
+    {
+        switch (action) {
+        case PM_DEVICE_ACTION_SUSPEND:
+            /* suspend the device */
+            ...
+            break;
+        case PM_DEVICE_ACTION_RESUME:
+            /* resume the device */
+            ...
+            break;
+        case PM_DEVICE_ACTION_TURN_ON:
+            /* powered on the device */
+            ...
+        case PM_DEVICE_ACTION_TURN_OFF:
+            /* powered off the device */
+            ...
+        default:
+            return -ENOTSUP;
+        }
+
+        return 0;
+    }
+
+    PM_DEVICE_DEFINE(dummy_driver, dummy_driver_pm_action);
+
+    DEVICE_DEFINE(dummy_driver, DUMMY_DRIVER_NAME, &dummy_init,
+        PM_DEVICE_GET(dummy_driver), NULL, NULL, APPLICATION,
+        CONFIG_KERNEL_INIT_PRIORITY_DEFAULT, &funcs);
```

Comment:
I would really like to see this reworked to use the devicetree variants. 99.9% of drivers are going to be using DT to initialize their devices.

Reply:
Yep, that makes sense. Changed.